src/components/Holder.js

```diff
@@ -6,9 +7,11 @@ import { TypeBox } from "./TypeBox";
 
 export const Holder = () => {
   const { pokemons } = useSelector((store) => store.pokemon);
-  const copyPokemons = [...pokemons];
-  const shuffledPokemons = copyPokemons.sort(() => 0.5 - Math.random());
-  const randomPokemons = shuffledPokemons.slice(0, 5);
+
+  const randomPokemons = useMemo(
+    () => pokemons.slice().sort(() => 0.5 - Math.random()).slice(0, 5),
```

Comment:
First, I wrapped the computed random pokemons list in a `useMemo` to avoid unnecessary re-renders and to avoid mutating the original store object.